opennlp-tools/src/main/java/opennlp/tools/ml/model/TwoPassDataIndexer.java

```diff
@@ -59,20 +62,28 @@ public void index(ObjectStream<Event> eventStream) throws IOException {
     File tmp = File.createTempFile("events", null);
     tmp.deleteOnExit();
     int numEvents;
-    try (Writer osw = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(tmp),
-        StandardCharsets.UTF_8))) {
-      numEvents = computeEventCounts(eventStream, osw, predicateIndex, cutoff);
+    BigInteger writeHash;
+    HashSumEventStream writeEventStream = new HashSumEventStream(eventStream);  // do not close.
+    try (DataOutputStream dos = new DataOutputStream(new BufferedOutputStream(new FileOutputStream(tmp)))) {
+      numEvents = computeEventCounts(writeEventStream, dos, predicateIndex, cutoff);
     }
+    writeHash = writeEventStream.calculateHashSum();
+
     display("done. " + numEvents + " events\n");
 
     display("\tIndexing...  ");
 
     List<ComparableEvent> eventsToCompare;
-    try (FileEventStream fes = new FileEventStream(tmp)) {
-      eventsToCompare = index(fes, predicateIndex);
+    BigInteger readHash = null;
+    try (HashSumEventStream readStream = new HashSumEventStream(new EventStream(tmp))) {
+      eventsToCompare = index(readStream, predicateIndex);
+      readHash = readStream.calculateHashSum();
     }
-
     tmp.delete();
+
+    if (readHash.compareTo(writeHash) != 0)
+      throw new RuntimeException("Event hash for writing and reading events did not match.");
```

Comment:
Seems nicer to me to throw here an IOException instead, after all this probably happens because of io issues.

Reply:
Maybe IllegalStateException?

Reply:
The cause for exception will be problems with writing and reading from disk. Why is IOException not the right one here?

Reply:
Got it. Thank you.